mlflow/pyfunc/__init__.py

```diff
@@ -395,10 +407,74 @@ def predict(self, data: PyFuncInput) -> PyFuncOutput:
 
         :param data: Model input as one of pandas.DataFrame, numpy.ndarray,
                      scipy.sparse.(csc.csc_matrix | csr.csr_matrix), List[Any], or
-                     Dict[str, numpy.ndarray]
+                     Dict[str, numpy.ndarray].
+                     For model signature with tensor spec inputs
+                    (e.g. the Tensorflow core / Keras model), the data must be one of
+                    numpy.ndarray, List[numpy.ndarray], Dict[str, numpy.ndarray], pandas.DataFrame,
+                    if data is pandas.DataFrame type, if a input field requires multidimensional
+                    array input, the corresponding column values in the pandas Dataframe will be
+                    reshaped to the required shape, and dataframe column values will be casted as
+                    the required tensor spec type.
+
         :return: Model predictions as one of pandas.DataFrame, pandas.Series, numpy.ndarray or list.
         """
         input_schema = self.metadata.get_input_schema()
+
+        if input_schema is not None and input_schema.is_tensor_spec():
+            if isinstance(data, pandas.DataFrame):
+                """
+                For tensor spec input schema, the model only accepts numpy array input or
```

Comment:
With the new strict pylint rules, inline block comments will need a "#" with no block quotes if not in typical pydoc locations I believe